File: services/mill-metadata-service/src/main/java/io/qpointz/mill/metadata/api/dto/TreeNodeDto.java
package io.qpointz.mill.metadata.api.dto;

import io.qpointz.mill.metadata.domain.MetadataType;
import lombok.Builder;
import lombok.Data;

import java.util.List;

/**
 * DTO for tree navigation responses.
 */
@Data
@Builder
public class TreeNodeDto {
    private String id;
    private String name;
    private MetadataType type;
    private String displayName;
    private String description;
    private List<TreeNodeDto> children;
    private boolean hasChildren;  // Hint for lazy loading
}

File: metadata/mill-metadata-service/src/main/kotlin/io/qpointz/mill/metadata/api/dto/TreeNodeDto.kt
package io.qpointz.mill.metadata.api.dto

data class TreeNodeDto(
    val id: String?,
    val name: String?,
    val kind: String?,
    val displayName: String?,
    val description: String? = null,
    var children: List<TreeNodeDto>? = null,
    var hasChildren: Boolean = false
)
